libs/algorithms/CMakeLists.txt

```diff
@@ -0,0 +1,125 @@
+  hpx/parallel/algorithms/transform_inclusive_scan.hpp
+  hpx/parallel/algorithms/transform.hpp
+  hpx/parallel/algorithms/transform_exclusive_scan.hpp
+  hpx/parallel/algorithms/swap_ranges.hpp
+  hpx/parallel/algorithms/sort_by_key.hpp
+  hpx/parallel/algorithms/set_union.hpp
+  hpx/parallel/algorithms/set_symmetric_difference.hpp
+  hpx/parallel/algorithms/set_intersection.hpp
+  hpx/parallel/algorithms/set_difference.hpp
+  hpx/parallel/algorithms/search.hpp
+  hpx/parallel/algorithms/rotate.hpp
+  hpx/parallel/algorithms/reverse.hpp
+  hpx/parallel/algorithms/replace.hpp
+  hpx/parallel/algorithms/remove.hpp
+  hpx/parallel/algorithms/remove_copy.hpp
+  hpx/parallel/algorithms/reduce.hpp
+  hpx/parallel/algorithms/reduce_by_key.hpp
+  hpx/parallel/algorithms/move.hpp
+  hpx/parallel/algorithms/mismatch.hpp
+  hpx/parallel/algorithms/lexicographical_compare.hpp
+  hpx/parallel/algorithms/is_sorted.hpp
+  hpx/parallel/algorithms/is_partitioned.hpp
+  hpx/parallel/algorithms/is_heap.hpp
+  hpx/parallel/algorithms/inclusive_scan.hpp
+  hpx/parallel/algorithms/includes.hpp
+  hpx/parallel/algorithms/generate.hpp
+  hpx/parallel/algorithms/for_loop_induction.hpp
+  hpx/parallel/algorithms/for_each.hpp
+  hpx/parallel/algorithms/find.hpp
+  hpx/parallel/algorithms/fill.hpp
+  hpx/parallel/algorithms/exclusive_scan.hpp
+  hpx/parallel/algorithms/equal.hpp
+  hpx/parallel/algorithm.hpp
+  hpx/parallel/algorithms/detail/transfer.hpp
+  hpx/parallel/algorithms/detail/is_negative.hpp
+  hpx/parallel/algorithms/detail/distance.hpp
+  hpx/parallel/algorithms/detail/dispatch.hpp
+  hpx/parallel/algorithms/detail/accumulate.hpp
+  hpx/parallel/algorithms/destroy.hpp
+  hpx/parallel/algorithms/count.hpp
+  hpx/parallel/algorithms/all_any_none.hpp
+  hpx/parallel/algorithms/adjacent_find.hpp
+  hpx/parallel/algorithms/adjacent_difference.hpp
+  hpx/parallel/util/partitioner_with_cleanup.hpp
+  hpx/parallel/util/detail/select_partitioner.hpp
+  hpx/parallel/util/detail/scoped_executor_parameters.hpp
+  hpx/parallel/util/detail/partitioner_iteration.hpp
+  hpx/parallel/util/detail/chunk_size_iterator.hpp
+  hpx/parallel/util/zip_iterator.hpp
+  hpx/parallel/util/transform_loop.hpp
+  hpx/parallel/util/transfer.hpp
+  hpx/parallel/util/projection_identity.hpp
+  hpx/parallel/util/prefetching.hpp
+  hpx/parallel/util/invoke_projected.hpp
+  hpx/parallel/util/detail/handle_remote_exceptions.hpp
+  hpx/parallel/util/detail/handle_local_exceptions.hpp
+  hpx/parallel/util/detail/algorithm_result.hpp
+  hpx/parallel/util/compare_projected.hpp
+  hpx/parallel/util/cancellation_token.hpp
+)
+
+include(HPX_AddModule)
+add_hpx_module(algorithms
+    DEPRECATION_WARNINGS
+    INSTALL_BINARIES OFF
+    FORCE_LINKING_GEN ON
+    HEADERS ${algorithms_headers}
+    # TODO: Really depends on main HPX library, but cannot add that as a
+    # dependency yet. Need to have the parts that algorithms depends on in
+    # separate modules first.
+    DEPENDENCIES
```

Comment:
Shouldn't this module depend on hpx_config?

Reply:
Most likely it depends on hpx_assert as well.

Reply:
Yes, you're right. Thanks!